### PR TITLE
updates for .NET 9

### PR DESCRIPTION
**Pull request by jennyf19**

### CODE REVIEW

File: .github/workflows/dotnetcore.yml [hunk 1]
```diff
@@ -22,14 +22,18 @@ jobs:
     runs-on: windows-latest
     continue-on-error: true
     name: "Build and run unit tests"
+
+    strategy: 
+      matrix:
+        dotnet-version:[ '6.0.x', '7.0.x', '8.0.x', '9.0.x' ]
```
Comment (jennyf19) [inline .github/workflows/dotnetcore.yml:28]: this seems to help against any issues w/the frameworks not running properly, there seemed to be an issue w/the wrong framework running on the OIDC tests

File: build/test-aot.ps1 [hunk 2]
```diff
@@ -16,7 +16,7 @@ foreach ($line in $($publishOutput -split "`r`n"))
 }
 
 Write-Host "Actual warning count is: ", $actualWarningCount
-$expectedWarningCount = 63
+$expectedWarningCount = 61
```
Comment (jennyf19) [inline build/test-aot.ps1:19]: if the count is lower, that is okay. we don't want it going up. I assume some dependencies were updated.

File: .github/workflows/dotnetcore.yml (hunk 1 above)
Comment (pmaytak) [reply .github/workflows/dotnetcore.yml:28]: I tried installing all frameworks yesterday, it didn't help. But if this works, then it works...

Comment (jennyf19) [reply .github/workflows/dotnetcore.yml:28]: it worked when I did a test PR

**Review by pmaytak — APPROVED** [review]

File: .github/workflows/dotnetcore.yml (hunk 1 above)
Comment (pmaytak) [reply .github/workflows/dotnetcore.yml:28]: The run failed?
https://github.com/AzureAD/microsoft-identity-web/actions/runs/11822912264

Comment (pmaytak) [reply .github/workflows/dotnetcore.yml:28]: The run failed?
https://github.com/AzureAD/microsoft-identity-web/actions/runs/11822912264

Comment (jennyf19) [reply .github/workflows/dotnetcore.yml:28]: That's not what it showed. and then I changed the base branch so I don't have proof.

**Pull request closed without merging**